### PR TITLE
tree2: Move node key support onto new API

### PR DESCRIPTION
**Pull request by CraigMacomber**
Node key scanning was one of the few (maybe only) internal uses of editable-tree in the package other than tests. This remove it (by porting it to the new API) as part of migrating off the old API.

This preserves a lot of the tests, but some will need updating.

As this feature is currently not ready for use and is not a priority, the perf tests were not ported to save time.

## Breaking Changes

Removes node key support from old editable-tree API, and adds it to the new editable-tree-2 API.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts [hunk 1]
```diff
@@ -697,6 +697,15 @@ export interface OptionalField<TTypes extends AllowedTypes> extends TreeField {
 	setContent(content: undefined | FlexibleNodeContent<TTypes>): void;
 }
 
+/**
+ * Field that an immutable {@link StableNodeKey} identifying this node.
```
Comment (noencke) [inline experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts:701]:
```suggestion
 * Field that contains an immutable {@link StableNodeKey} identifying this node.
```

**Review by noencke — APPROVED** [review]
Thanks for doing this. Moving the node key index out of SharedTree itself and into the EditableTree may not be what we want long term, but since node keys are not a priority I'm fine with this for now. We're going to need to do some re-architecting regardless.

**Review by noencke — COMMENTED** [review]
edit: Github's reviewing UI is impressively confusing considering how few capabilities it has.

**Comment by CraigMacomber** [comment]
> Thanks for doing this. Moving the node key index out of SharedTree itself and into the EditableTree may not be what we want long term, but since node keys are not a priority I'm fine with this for now. We're going to need to do some re-architecting regardless.

Github bugged and shows this PR as approved without showing your feedback until the page refreshed after merging it. I'll include your comment fix in my next PR!

**Pull request merged**